ts/src/binance.ts

```diff
@@ -2853,12 +2878,23 @@ export default class binance extends Exchange {
                 throw new ExchangeError (this.id + ' fetchMarkets() this.options fetchMarkets "' + marketType + '" is not a supported market type');
             }
         }
-        const promises = await Promise.all (promisesRaw);
+        const results = await Promise.all (promisesRaw);
         let markets = [];
+        this.options['crossMarginPairsData'] = {};
+        this.options['isolatedMarginPairsData'] = {};
         for (let i = 0; i < fetchMarkets.length; i++) {
-            const promise = this.safeDict (promises, i);
-            const promiseMarkets = this.safeList2 (promise, 'symbols', 'optionSymbols', []);
-            markets = this.arrayConcat (markets, promiseMarkets);
+            const res = this.safeValue (results, i);
+            if (fetchMargins && Array.isArray (res)) {
+                // first one is the cross-margin promise
+                if (this.options['crossMarginPairsData'] === undefined) {
+                    this.options['crossMarginPairsData'] = this.indexBy (res, 'symbol');
```

Comment:
we're indexing the entire response and then we just check the symbol

Reply:
wdym here? i dont find out the issue

Reply:
@ttodua you do `this.indexBy (res, 'symbol')` and then `id in this.options['crossMarginPairsData']` meaning you only need the symbol used as `key` and not the entire response

Reply:
ah, i see. fixed.